<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>can_terminal</class>
 <widget class="QWidget" name="can_terminal">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1296</width>
    <height>649</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="handleWidth">
      <number>10</number>
     </property>
     <property name="childrenCollapsible">
      <bool>true</bool>
     </property>
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>CAN Terminal</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="0" colspan="3">
        <widget class="QSplitter" name="splitter_2">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <widget class="QTextEdit" name="terminal">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="font">
           <font>
            <family>Monospace</family>
           </font>
          </property>
          <property name="focusPolicy">
           <enum>Qt::NoFocus</enum>
          </property>
          <property name="lineWrapMode">
           <enum>QTextEdit::NoWrap</enum>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
         <widget class="QGroupBox" name="groupBox_3">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Devices</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_5">
           <item row="1" column="0">
            <widget class="QPushButton" name="devNew">
             <property name="text">
              <string>New</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QPushButton" name="devDel">
             <property name="text">
              <string>Delete</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0" colspan="2">
            <widget class="QTableWidget" name="devs">
             <property name="minimumSize">
              <size>
               <width>100</width>
               <height>0</height>
              </size>
             </property>
             <attribute name="horizontalHeaderStretchLastSection">
              <bool>true</bool>
             </attribute>
             <column>
              <property name="text">
               <string>ID</string>
              </property>
             </column>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QPushButton" name="terminal_clear">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>Clear</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLineEdit" name="terminal_cmd"/>
       </item>
       <item row="2" column="1">
        <widget class="QPushButton" name="terminal_send">
         <property name="focusPolicy">
          <enum>Qt::TabFocus</enum>
         </property>
         <property name="text">
          <string>Send</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QRadioButton" name="extProt">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Uses the older UV terminal protocol with extended message frames. The device terminal communication is made by sending an extended message with an ID of 0x15560000 + node_ID.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>Extended protocol</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QRadioButton" name="sdoProt">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Uses the newer CANopen SDO-based UV terminal protocol for communicating with the devices. Commands are sent to devices as SDO write requests to index 0x5FFF and subindex 0. The device responds as a SDO write response from the same subindex.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>SDO protocol</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>Can</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="2" column="0">
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="1" column="1">
          <widget class="QLineEdit" name="canData">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>180</width>
             <height>0</height>
            </size>
           </property>
           <property name="font">
            <font>
             <family>Monospace</family>
            </font>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Data (hexadecimal)</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>ID</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2" colspan="2">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Message type</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QRadioButton" name="canSTD">
           <property name="text">
            <string>STD</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="3">
          <widget class="QRadioButton" name="canEXT">
           <property name="text">
            <string>EXT</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QSpinBox" name="canID">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>536870912</number>
           </property>
           <property name="displayIntegerBase">
            <number>16</number>
           </property>
          </widget>
         </item>
         <item row="1" column="5">
          <widget class="QPushButton" name="canSend">
           <property name="focusPolicy">
            <enum>Qt::TabFocus</enum>
           </property>
           <property name="text">
            <string>Send</string>
           </property>
          </widget>
         </item>
         <item row="1" column="4">
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="1" column="6">
          <widget class="QPushButton" name="canClear">
           <property name="text">
            <string>Clear</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="0" column="0">
        <widget class="QTableWidget" name="can_table">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="focusPolicy">
          <enum>Qt::ClickFocus</enum>
         </property>
         <attribute name="horizontalHeaderStretchLastSection">
          <bool>true</bool>
         </attribute>
         <attribute name="verticalHeaderDefaultSectionSize">
          <number>18</number>
         </attribute>
         <column>
          <property name="text">
           <string>Time</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>ID</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Type</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Len</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Data</string>
          </property>
         </column>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>devNew</tabstop>
  <tabstop>devDel</tabstop>
  <tabstop>devs</tabstop>
  <tabstop>terminal_cmd</tabstop>
  <tabstop>terminal_send</tabstop>
  <tabstop>terminal_clear</tabstop>
  <tabstop>canID</tabstop>
  <tabstop>canData</tabstop>
  <tabstop>canSTD</tabstop>
  <tabstop>canEXT</tabstop>
  <tabstop>canSend</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>terminal_cmd</sender>
   <signal>returnPressed()</signal>
   <receiver>terminal_send</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>317</x>
     <y>619</y>
    </hint>
    <hint type="destinationlabel">
     <x>370</x>
     <y>619</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>terminal_clear</sender>
   <signal>clicked()</signal>
   <receiver>terminal_cmd</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>478</x>
     <y>622</y>
    </hint>
    <hint type="destinationlabel">
     <x>170</x>
     <y>621</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>terminal_clear</sender>
   <signal>clicked()</signal>
   <receiver>terminal</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>501</x>
     <y>615</y>
    </hint>
    <hint type="destinationlabel">
     <x>237</x>
     <y>533</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
